integrations/terraform-cloud/client.py

```diff
@@ -0,0 +1,174 @@
+
+    async def get_paginated_resources(
+        self, endpoint: str, params: Optional[dict[str, Any]] = None
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        params = params or {}
+        params.update({"page[size]": PAGE_SIZE})
+        next_url = endpoint
+
+        while next_url:
+            response = await self.send_api_request(endpoint=next_url, query_params=params)
+            resources = response.get("data", [])
+
+            logger.info(f"Fetched {len(resources)} resources from {next_url}")
+            yield resources
+
+            next_url = response.get("links", {}).get("next", "")
+            if not next_url:
+                logger.debug(f"No more pages to fetch for {endpoint}")
+                break
+
+    async def get_paginated_organizations(self) -> list[dict[str, Any]]:
+        logger.info("Fetching organizations")
+        async for organizations in self.get_paginated_resources("organizations"):
+            yield organizations
+
+    async def get_single_workspace(self, workspace_id: str) -> dict[str, Any]:
+        logger.info(f"Fetching workspace with ID: {workspace_id}")
+        workspace = await self.send_api_request(endpoint=f"workspaces/{workspace_id}")
+        return workspace.get("data")
+
+    async def get_single_run(self, run_id: str) -> dict[str, Any]:
+        logger.info(f"Fetching run with ID: {run_id}")
+        run = await self.send_api_request(endpoint=f"runs/{run_id}")
+        return run.get("data",{})
+
+    async def get_state_version_output(self, state_version_id: str) -> dict[str, Any]:
+        logger.info(f"Fetching state version output for ID: {state_version_id}")
+        outputs = await self.send_api_request(endpoint=f"state-versions/{state_version_id}/outputs")
+        return outputs.get("data",{})
+
+    async def get_paginated_workspaces(
+        self,
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        cache_key = "terraform-cloud-workspace"
+        if cache := event.attributes.get(cache_key):
+            logger.info("Retrieving workspaces data from cache")
+            yield cache
+            return
+
+        all_workspaces = []
+        async for organizations in self.get_paginated_organizations():
+            for organization in organizations:
+                organization_id = organization['id']
+                endpoint = f"organizations/{organization_id}/workspaces"
+                async for workspaces in self.get_paginated_resources(endpoint):
+                    all_workspaces.extend(workspaces)
+                    yield workspaces
+
+        event.attributes[cache_key] = workspaces
+
+
+    async def get_paginated_runs_for_workspace(
+        self, workspace_id: str
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        logger.info(f"Fetching runs for workspace ID: {workspace_id}")
+
+        endpoint = f"workspaces/{workspace_id}/runs"
+
+        async for runs in self.get_paginated_resources(endpoint):
+            yield runs

```

Comment:
I'm unable to work with get_paginated_runs if i use the return approach, the suggested changes, throws this error.
I think because get_paginated_resources is an asynchronous generator function. I can't simply return its value from another async function because it's designed to yield multiple values over time, not return a single awaitable result.

I get this error if attempt `return self.get_paginated_resources(endpoint)` 
/home/mk-armah/SAVANNAHTECH/ocean/integrations/terraform-cloud/./main.py:67: RuntimeWarning: coroutine 'TerraformClient.get_paginated_runs_for_workspace' was never awaited
  return [
RuntimeWarning: Enable tracemalloc to get the object allocation traceback

Then i tried return `await self.get_paginated_resources(endpoint)` .
which also produces the same error


I'd appreciate it if an existing working example is shared